frontend/app/components/session-timeout.tsx

```diff
@@ -0,0 +1,113 @@
+    // Disable default event listeners; The IdleTimer should only activate during route navigation and form
+    // submissions, as these actions will interact with the session and extend its lifespan.
+    events: [],
+    onIdle: async () => {
+      // Trigger the session end function when the user becomes idle.
+      await onSessionEnd();
+    },
+    promptBeforeIdle,
+    timeout,
+  });
+
+  // Track the current location and its key for detecting navigation changes.
+  const { key: locationKey } = useLocation();
+
+  useEffect(() => {
+    // Activate the IdleTimer whenever the location changes (indicating a route navigation).
+    activate();
+  }, [locationKey, activate]);
+
+  // Track fetcher states to activate the IdleTimer during submissions.
+  const fetchers = useFetchers();
+  const fetcherSubmitting = fetchers.filter(({ state }) => state === 'submitting').length > 0;
+
+  useEffect(() => {
+    // Activate the IdleTimer if any fetcher is submitting.
+    if (fetcherSubmitting) {
+      activate();
+    }
+  }, [fetcherSubmitting, activate]);
+
+  // Track Remix's <Form> submission state to activate the IdleTimer during <Form> submissions.
+  const navigation = useNavigation();
+  const formSubmitting = navigation.state === 'submitting';
+
+  useEffect(() => {
+    // Activate the IdleTimer if a <Form> is submitting.
+    if (formSubmitting) {
+      activate();
+    }
+  }, [formSubmitting, activate]);
+
+  useEffect(() => {
+    const updateRemainingTime = () => {
+      const remainingTime = getRemainingTime();
+      const minutes = Math.floor(remainingTime / 60_000);
+      const seconds = Math.floor((remainingTime % 60_000) / 1_000);
+      const formattedTime = `${minutes}:${seconds.toString().padStart(2, '0')}`;
+      setTimeRemaining(formattedTime);
+    };
+
+    const interval = setInterval(updateRemainingTime, 1000);
+    updateRemainingTime(); // Initial call to set the time immediately
+
+    return () => {
+      clearInterval(interval);
+    };
+  }, [getRemainingTime]);
+
+  return (
+    <Dialog open={isPrompted()} onOpenChange={async (open) => !open && (await onSessionExtend())}>
+      <DialogContent aria-describedby={undefined} className="sm:max-w-md">
+        <DialogHeader>
+          <DialogTitle>{t('gcweb:session-timeout.header')}</DialogTitle>
+        </DialogHeader>
+        {t('gcweb:session-timeout.description', { timeRemaining })}
+        <DialogFooter>
+          <Button id="end-session-button" variant="default" size="sm" onClick={async () => await onSessionEnd()}>
+            {t('gcweb:session-timeout.end-session')}
+          </Button>
+          <Button id="continue-session-button" variant="primary" size="sm" onClick={async () => await onSessionExtend()}>
+            {t('gcweb:session-timeout.continue-session')}
+          </Button>
+        </DialogFooter>
+      </DialogContent>
+    </Dialog>
+  );
+};
+
+export default SessionTimeout;
```

Comment:
use `export function SessionTimeout()` instead of exporting at the end of the file.